Fix some minor warnings from cppcheck in WTF

cdumez opened the pull request:
#### 9dd212febdb2ba8bc9900ed5eb2789a6ae72529e
<pre>
Fix some minor warnings from cppcheck in WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=254856">https://bugs.webkit.org/show_bug.cgi?id=254856</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
* Source/WTF/wtf/HexNumber.h:
(WTF::StringTypeAdapter&lt;HexNumberBuffer&gt;::StringTypeAdapter):
* Source/WTF/wtf/JSONValues.cpp:
* Source/WTF/wtf/MediaTime.cpp:
(WTF::greatestCommonDivisor):
* Source/WTF/wtf/ObjectIdentifier.cpp:
(WTF::ObjectIdentifierBase::generateIdentifierInternal):
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::StringTypeAdapter&lt;ObjectIdentifier&lt;T&gt;&gt;::StringTypeAdapter):
* Source/WTF/wtf/URLParser.cpp:
(WTF::pow256):
(WTF::URLParser::parseIPv4PieceInsideIPv6):
* Source/WTF/wtf/Vector.h:
(WTF::VectorBuffer::VectorBuffer):
* Source/WTF/wtf/dtoa.cpp:
(WTF::truncateTrailingZeros):
* Source/WTF/wtf/dtoa/fast-dtoa.cc:
* Source/WTF/wtf/dtoa/strtod.cc:
* Source/WTF/wtf/text/StringBuilderJSON.cpp:
(WTF::appendQuotedJSONStringInternal):
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::equal):
* Source/WTF/wtf/text/icu/UTextProviderLatin1.cpp:
(WTF::uTextLatin1Clone):
(WTF::openLatin1UTextProvider):
* Source/WTF/wtf/unicode/UTF8Conversion.cpp:
(WTF::Unicode::convertLatin1ToUTF8):
(WTF::Unicode::convertUTF16ToUTF8):
(WTF::Unicode::convertUTF8ToUTF16Impl):
(WTF::Unicode::convertUTF8ToUTF16):
(WTF::Unicode::convertUTF8ToUTF16ReplacingInvalidSequences):
* Source/WTF/wtf/unicode/UTF8Conversion.h:

Canonical link: <a href="https://commits.webkit.org/262482@main">https://commits.webkit.org/262482@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a71eb857e7700459f679ed2e554df98b5611a2a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1603 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1634 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1691 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2524 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1723 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1699 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1696 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1507 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1617 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1454 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1453 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2361 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1453 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1439 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1352 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1361 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1461 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1480 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2529 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/1540 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1497 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1340 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/1660 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1413 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1446 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/356 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/405 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1560 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/1700 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/471 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/MediaTime.cpp:
```diff
@@ -49,9 +49,8 @@ static uint32_t greatestCommonDivisor(uint32_t a, uint32_t b)
-        temp = b;
+        uint32_t temp = b;
         b = a % b;
         a = temp;
```
A C++ way to write this is:
```
b = std::exchange(a, b) % b;
````

darinadler commented on Source/WTF/wtf/ObjectIdentifier.cpp:
```diff
@@ -31,7 +31,7 @@ namespace WTF {
 uint64_t ObjectIdentifierBase::generateIdentifierInternal()
 {
-    static uint64_t current;
+    static uint64_t current = 0;
```
This is a really peculiar warning. Why does `cppcheck` care about this?

cdumez replied:
All it said is:
```
Source/WTF/wtf/ObjectIdentifier.cpp:34:21: style: Variable 'current' is not assigned a value. [unassignedVariable]
    static uint64_t current;
```

I would have expected it to get zero-initialized. It is possible this is a false positive.

cdumez committed:
Fix some minor warnings from cppcheck in WTF
https://bugs.webkit.org/show_bug.cgi?id=254856

Reviewed by Darin Adler.

* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
* Source/WTF/wtf/HexNumber.h:
(WTF::StringTypeAdapter<HexNumberBuffer>::StringTypeAdapter):
* Source/WTF/wtf/JSONValues.cpp:
* Source/WTF/wtf/MediaTime.cpp:
(WTF::greatestCommonDivisor):
* Source/WTF/wtf/ObjectIdentifier.cpp:
(WTF::ObjectIdentifierBase::generateIdentifierInternal):
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::StringTypeAdapter<ObjectIdentifier<T>>::StringTypeAdapter):
* Source/WTF/wtf/URLParser.cpp:
(WTF::pow256):
(WTF::URLParser::parseIPv4PieceInsideIPv6):
* Source/WTF/wtf/Vector.h:
(WTF::VectorBuffer::VectorBuffer):
* Source/WTF/wtf/dtoa.cpp:
(WTF::truncateTrailingZeros):
* Source/WTF/wtf/dtoa/fast-dtoa.cc:
* Source/WTF/wtf/dtoa/strtod.cc:
* Source/WTF/wtf/text/StringBuilderJSON.cpp:
(WTF::appendQuotedJSONStringInternal):
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::equal):
* Source/WTF/wtf/text/icu/UTextProviderLatin1.cpp:
(WTF::uTextLatin1Clone):
(WTF::openLatin1UTextProvider):
* Source/WTF/wtf/unicode/UTF8Conversion.cpp:
(WTF::Unicode::convertLatin1ToUTF8):
(WTF::Unicode::convertUTF16ToUTF8):
(WTF::Unicode::convertUTF8ToUTF16Impl):
(WTF::Unicode::convertUTF8ToUTF16):
(WTF::Unicode::convertUTF8ToUTF16ReplacingInvalidSequences):
* Source/WTF/wtf/unicode/UTF8Conversion.h:

Canonical link: https://commits.webkit.org/262482@main
